cmd/load-runner/runner.go

```diff
@@ -99,3 +121,45 @@ func (r *LoadTestRunner) createValuesFileCopy(ctx context.Context) error {
 	r.logger.Infon("Successfully created values copy file", logger.NewStringField("file", copyFile))
 	return nil
 }
+
+func (r *LoadTestRunner) startPortForward(ctx context.Context, namespace string) (func(), error) {
+	if err := r.portForward.Start(ctx, namespace); err != nil {
+		return nil, fmt.Errorf("failed to start port-forward: %w", err)
+	}
+	stopPortForward := func() {
+		if err := r.portForward.Stop(); err != nil {
+			r.logger.Errorn("Failed to stop port-forward", obskit.Error(err))
+		}
+	}
+
+	return stopPortForward, nil
+}
+
+func (r *LoadTestRunner) monitorMetrics(ctx context.Context, startTime time.Time) {
+	if r.config.Reporting.Interval == "" {
+		return
+	}
+
+	reportPeriod, err := time.ParseDuration(r.config.Reporting.Interval)
+	if err != nil {
+		r.logger.Errorn("Invalid monitoring report period, using default of 10s", obskit.Error(err))
+		reportPeriod = 10 * time.Second
+	}
+
+	ticker := time.NewTicker(reportPeriod)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-ticker.C:
+			metrics, err := r.mimirClient.GetMetrics(ctx, r.config.Reporting.Metrics)
+			if err != nil {
+				r.logger.Errorn("Failed to get current metrics", obskit.Error(err))
+				continue
+			}
+			r.logger.Infon("", logger.NewStringField("metrics", fmt.Sprintf("%+v", metrics)))
```

Comment:
The whole point of using a non-sugared logger is to avoid reflection. Also please avoid logging without a message (i.e. `""`). If you really can't avoid reflection you can either use `logger.NewField()` along with the non-sugared logger or `Infow` for using structured logs with reflection. Beware that `Infow` might be flagged in the future by linters.

Reply:
Got it. Updated the logic to print the metrics.